### ISSUE TITLE
[Feature]: Add GPEN to Face Restoration models

### ISSUE THREAD

**Issue by not-ski** [Gourieff/sd-webui-reactor#433]
### Feature description

I have tested many face restoration models now (including CodeFormer, GFPGAN, RestoreFormer++, DMDNet, and GPEN) and GPEN has been the best-performing _by far_. IMO, it outpaces the options currently available (CodeFormer and GFPGAN) by a considerable margin.

GPEN repo can be found here: https://github.com/yangxy/GPEN

It also seems that FaceFusion have made the GPEN models available in ONNX form as well: https://github.com/facefusion/facefusion-assets

The roop-unleashed implementation of GPEN could also potentially be useful: https://github.com/C0untFloyd/roop-unleashed/blob/main/roop/processors/Enhance_GPEN.py

Thanks <3

**Comment by Gourieff** [comment]
Yes, this is in the roadmap
But already now you can use ReActor with GPEN via ComfyUI https://github.com/Gourieff/comfyui-reactor-node
ReActor Node supports [GPEN-BFR-512 ONNX model](https://huggingface.co/datasets/Gourieff/ReActor/blob/main/models/facerestore_models/GPEN-BFR-512.onnx)

**Comment by Pandasarethebest93** [comment]
> Owner
> ###

When do you think that will be implemented. I use it in comfyui currently however it would be ideal to have available for when I generate things in A1111 since it isn't possible to recreate the exact same images in comfy And it is time consuming to import the image into comfy after generating in a1111 when I want to generate a batch and have them faceswap and upscale. Much appreciated for you creating this extension.

**Comment by web2299** [comment]
Yes i am agree adding the GPEN-BFR models support to the Reactor automatic1111 extension will be VERY useful ! I am waiting for this for long time ...

**Comment by juandiLiza** [comment]
hi @Gourieff I have installed ReActor in ComfyUI but I dont have the [GPEN-BFR-512 ONNX model] or 1024 or 2048, how can I add them?

![image](https://github.com/user-attachments/assets/e2b52d13-dbc2-4ef4-a3c4-26e1cc29392c)

**Comment by morisuba** [comment]
> Yes, this is in the roadmap But already now you can use ReActor with GPEN via ComfyUI https://github.com/Gourieff/comfyui-reactor-node ReActor Node supports [GPEN-BFR-512 ONNX model](https://huggingface.co/datasets/Gourieff/ReActor/blob/main/models/facerestore_models/GPEN-BFR-512.onnx)

any update on this ?

**Comment by vinch00** [comment]
@Gourieff would highly appreciate GPEN in A1111 extension. It will make ReActor so much better from where it already is!

**Comment by Bananenbrot86** [comment]
Hi @juandiLiza I have the same problem. Have you found a solution yet? I only have two options: GFPGANv1.3 and 1.4, and 1.4 doesn't even work due to an error: ReActorFaceSwap unexpected EOF, expected 2054890 more bytes. The file might be corrupted. I've tried everything possible to solve this, without success.
